app.ts

```diff
@@ -0,0 +1,309 @@
+		numOfRows = 8;
+	} else if ((screenHeight < 265) && (screenHeight >= 95)) {
+		numOfRows = 3;
+	} else if (screenHeight < 95) {
+		numOfRows = 2;
+	}
+
+	// Recreate grid with new set number of rows
+	createInitialPage(numOfRows);
+});
+
+// Set a timeout to use for promises when navigating through data
+let clickTimeout: ReturnType<typeof setTimeout>;
+
+// When clicking on left arrow
+$( "#leftarrow" ).on( "click", function() {
+	// Clear the timeout every time if you navigate and you're not at the start ID of all the data
+	if (fromID != 1) {
+		clearTimeout(clickTimeout);
+	}
+
+	// Clear form value in front-end when navigating data using the arrows
+	let startfrom = document.querySelector("#startfrom") as HTMLInputElement;
+	startfrom.value = "";
+	
+	if ((fromID < (numOfRows-1)) && (fromID > 1)) {		// If the "from" ID is within the IDs at the beginning of the dataset
+		fromID = 1;
+		toID = numOfRows;
+		fromIDElement.innerHTML = fromID.toString();
+		toIDElement.innerHTML = toID.toString();
+
+		// Create table using the new set IDs inside a set timeout
+		clickTimeout = setTimeout(function(){
+			// Use resolved promises to fetch data from backend
+			let headings = Promise.resolve(fetch("/columns").then(res => res.text())); 
+			let records = Promise.resolve(fetch("/records?from="+fromID+"&to="+toID).then(res => res.text()));
+			// Promise.all<string>([headings, records]).then((values) => {
+			Promise.all<string>([headings, records]).then((values) => {
+				generateTable(values[0], values[1]);
+			}).catch(err => console.log(err));
+		}, 500);
+	} else if (fromID === 1) {							// If the "from" ID is 1 then set the values and do not make any backend calls
+		fromID = 1;
+		toID = numOfRows;
+		fromIDElement.innerHTML = fromID.toString();
+		toIDElement.innerHTML = toID.toString();
+	} else {											// Else, continue creating table with the set IDs
+		fromID = fromID - (numOfRows-1);
+		toID = toID - (numOfRows-1);
+		fromIDElement.innerHTML = fromID.toString();
+		toIDElement.innerHTML = toID.toString();
+
+		// Create table using the new set IDs inside a set timeout
+		clickTimeout = setTimeout(function(){
+			// Use resolved promises to fetch data from backend
+			let headings = Promise.resolve(fetch("/columns").then(res => res.text()));
+			let records = Promise.resolve(fetch("/records?from="+fromID+"&to="+toID).then(res => res.text()));
+			Promise.all<string>([headings, records]).then((values) => {
+				generateTable(values[0], values[1]);
+			}).catch(err => console.log(err));
+		}, 500);
+	}
+});
+					
+// Listen for click on right arrow
+$( "#rightarrow" ).on( "click", function() {
+	// Clear the timeout every time if you navigate and you're not at the end ID of all the data
+	if (toID != totalNumofRecords) {
+			clearTimeout(clickTimeout);
+	}
+
+	// Clear form value in front-end when navigating data using the arrows
+	let startfrom = document.querySelector("#startfrom") as HTMLInputElement;
```

Comment:
I also feel these two lines get repeated a lot and can be moved into an independent method